File: KoRadio/KoRadio.Services/Database/Freelancer.cs
using System;
using System.Collections.Generic;

namespace KoRadio.Services.Database;

public partial class Freelancer:ISoftDelete,IApplicantDelete
{
    public int FreelancerId { get; set; }

    public string Bio { get; set; } = null!;

    public decimal Rating { get; set; }

    public int ExperianceYears { get; set; }

    public int WorkingDays { get; set; }

    public TimeOnly StartTime { get; set; }

    public TimeOnly EndTime { get; set; }

    public bool IsDeleted { get; set; }

    public bool IsApplicant { get; set; }

    public int? TotalRatings { get; set; }

    public double? RatingSum { get; set; }

    public virtual User FreelancerNavigation { get; set; } = null!;

    public virtual ICollection<FreelancerService> FreelancerServices { get; set; } = new List<FreelancerService>();

    public virtual ICollection<Job> Jobs { get; set; } = new List<Job>();

    public virtual ICollection<TenderBid> TenderBids { get; set; } = new List<TenderBid>();

    public virtual ICollection<Tender> Tenders { get; set; } = new List<Tender>();

    public virtual ICollection<UserRating> UserRatings { get; set; } = new List<UserRating>();
}

File: KoRadio/UI/ko_radio_mobile/lib/models/freelancer.dart


import 'package:json_annotation/json_annotation.dart';
import 'package:ko_radio_mobile/models/freelancer_service.dart';
import 'package:ko_radio_mobile/models/user.dart';


part 'freelancer.g.dart';
@JsonSerializable()
class Freelancer{
  int freelancerId;
  String bio;
  double rating;
  int experianceYears;
  List<String> workingDays;
  String startTime;
  String endTime;
  List<FreelancerService> freelancerServices;
  User freelancerNavigation;


  
  Freelancer({required this.bio, required this.rating,  required this.experianceYears,required this.freelancerServices,required this.workingDays, required this.startTime, required this.endTime, required this.freelancerNavigation, required this.freelancerId});

  factory Freelancer.fromJson(Map<String, dynamic> json) => _$FreelancerFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$FreelancerToJson(this);
}
